internal/resource/config/datastore/custom_data_store.go

```diff
@@ -57,7 +59,20 @@ func toSchemaCustomDataStore() schema.SingleNestedAttribute {
 		"plugin_descriptor_ref": schema.SingleNestedAttribute{
 			Required:    true,
 			Description: "Reference to the plugin descriptor for this instance. The plugin descriptor cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
-			Attributes:  resourcelink.ToSchema(),
+			Attributes: map[string]schema.Attribute{
```

Comment:
You could move this to another method in the resourcelink package so that the duplication is limited to one place

Reply:
moved to common package